### ISSUE TITLE
Lack of color contrast for accessibility

### ISSUE THREAD

**Issue by edgarasben** [tailwindlabs/tailwindcss#3876]
### What version of Tailwind CSS are you using?

v2.0.4

### What version of Node.js are you using?

v12.0.0

### What browser are you using?

Safari

### What operating system are you using?

macOS

### Reproduction repository

-

When working on some designs I noticed that some colors (orange, amber, yellow and lime) have a lower than 4.5:1 contrast ratio at 600 variation, compared to other colors. My expectation was that all colors at 600 would have at least 4.5:1 contrast ratio. This would ensure that picking **any** color at 600 would constantly give me an accessible color (for example for links on white background).

What do you think?

![image](https://user-images.githubusercontent.com/725324/112966172-42d82100-914a-11eb-8017-9f69dd969d55.png)

**Comment by adamwathan** [comment]
We tried to build a perceptually uniform color palette originally but the results are just always bad. Yellow is brighter than blue, that's just a characteristic of yellow and what makes it yellow. Trying to make yellow-400 have the same brightness as blue-400 means every yellow is basically just brown.

I would recommend just picking color combinations that meet your desired contrast ratio on a case-by-case basis instead of trying to build a rigid system of rules and try to get it to apply to all situations. It's perfectly straightforward to build 100% WCAG compliant designs with the default colors, you just need to be intentional about it. Trying to force everything into an overly-robotic system just doesn't give good results in my experience.

**Comment by edgarasben** [comment]
Thank you for you input Adam. Yes, I am aware of the fundamental color differences (as in yellow always perceived as brighter color). I just think the naming could help the semantics here. It would be nice that colors were adjusted in a way so I can pick 600 variable and I can be sure that it's always good for items like links where contrast is needed. Unless there is a general disagreement about 4.5:1 rule or it's hard to change the color system at this point.

**Comment by adamwathan** [comment]
If we did that we'd have to have different numbers for each color right, like yellow would be 100-900, blue would be 400-1200 or something since it's inherently darker, etc. Definitely can't change it regardless without it being a 3.0 release since it would be a breaking change but don't really want to either. I would just recommend picking the darker shades when working with lighter colors, or building your own color palette. I just really fundamentally disagree with the concept of perceptually uniform color palettes.

**Comment by edgarasben** [comment]
Yeah, either that or yellow fade to darker a bit sooner so it's 4.5:1 at 600. But I understand there are multiple side effects to this and possible backwards compatibility issues. 

This was more of an idea to have some kind of a semantic way to help people choosing the brightest yet accessible color. Maybe there are other more elegant ways of achieving the same goal.

I will leave with this link: https://colorbox.io. I like the way it shows with an arrow where the 4.5:1 starts on the color scale. Maybe something useful for the future. (I know you've tried many tools already and nothing is ideal out there).
